### PR TITLE
Perfective updates to HTML

### PR DESCRIPTION
**Pull request by eddieantonio**
Just making a few updates to the HTML that make me and the [Nu HTML checker](https://validator.w3.org/nu/) happy 😄 

There's some updated a11y here, so it actually does make things better!

### CODE REVIEW

File: CreeDictionary/CreeDictionary/templates/CreeDictionary/base.html [hunk 1]
```diff
@@ -68,13 +68,13 @@
     {% endblock %}
   </div>
 
-  <section class="search-results">
+  <div class="search-results">
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/templates/CreeDictionary/base.html:71]: This isn't really a document section, and validators complain if a `<section>` does not have a heading tag (e.g., `<h2>`, `<h3>`) within it.

File: CreeDictionary/CreeDictionary/templates/CreeDictionary/components/header.html [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
   <header class="branding top-bar__logo"><a href="{% url 'cree-dictionary-index' %}">
       <h1 class="branding__heading branding__title">{% orth 'itwêwina' %}</h1>
-      <p class="branding__heading branding__subtitle">Plains Cree Dictionary</p>
+      <p class="branding__heading branding__subtitle" role="doc-subtitle">Plains Cree Dictionary</p>
   </a></header>
-  {# TODO: The language and orthography selectors are not ready yet, so hide them: #}
+
   <nav class="top-bar__nav">
     <details class="toggle-box toggle-box--with-menu">
       <summary id="language-selector__button" class="toggle-box__toggle" data-cy="language-selector"
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/templates/CreeDictionary/components/header.html:11]: `<summary>` tags inherintly have the [ARIA role=button](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary#:~:text=Implicit%20ARIA%20role)!

File: CreeDictionary/CreeDictionary/templates/CreeDictionary/components/header.html [hunk 3]
```diff
@@ -3,13 +3,13 @@
...
-               aria-haspopup="menu" role="button" tabindex="0"
+               aria-haspopup="menu" tabindex="0"
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/templates/CreeDictionary/components/header.html:12]: `<summary>` tags inherently have the [ARIA role=button](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary#:~:text=Implicit%20ARIA%20role)!

File: CreeDictionary/CreeDictionary/templates/CreeDictionary/components/search.html [hunk 4]
```diff
@@ -9,6 +9,14 @@
...
              placeholder="Search in Cree or English"
+             {% comment %}
+             autocorrect="off" is a Safari extension that disables...
+             autocorrect! This is necessary since autocorrect often tries to
+             turn Cree words into English words, which is no bueno.
+
+             See: https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/Attributes.html#//apple_ref/doc/uid/TP40008058-autocorrect
+
+             {% endcomment %}
              autocapitalize="off" autocomplete="off" autocorrect="off" spellcheck="false"
```
Comment (Madoshakalaka) [inline CreeDictionary/CreeDictionary/templates/CreeDictionary/components/search.html:20]: dumb conputar, dumb

Comment (eddieantonio) [reply CreeDictionary/CreeDictionary/templates/CreeDictionary/components/search.html:20]: yeah, I hate this dumb, non-standard extension, but I hate iphone autocorrecting Cree into English even more!

**Pull request merged**